File: bindings/lua/Polycode/GlyphData.lua
class "GlyphData"


function GlyphData:__getvar(name)
	if name == "num_glyphs" then
		local retVal = Polycode.GlyphData_get_num_glyphs(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["FT_UInt"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "trailingAdvance" then
		return Polycode.GlyphData_get_trailingAdvance(self.__ptr)
	end
end

function GlyphData:__setvar(name,value)
	if name == "num_glyphs" then
		Polycode.GlyphData_set_num_glyphs(self.__ptr, value.__ptr)
		return true
	elseif name == "trailingAdvance" then
		Polycode.GlyphData_set_trailingAdvance(self.__ptr, value)
		return true
	end
	return false
end
function GlyphData:GlyphData(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.GlyphData(unpack(arg))
	end
end

function GlyphData:clearData()
	local retVal =  Polycode.GlyphData_clearData(self.__ptr)
end

function GlyphData:__delete()
	if self then Polycode.delete_GlyphData(self.__ptr) end
end

File: bindings/javascript/Polycode/GlyphData.js
function GlyphData() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.GlyphData()
	}
	Object.defineProperties(this, {
		'num_glyphs': { enumerable: true, configurable: true, get: GlyphData.prototype.__get_num_glyphs, set: GlyphData.prototype.__set_num_glyphs},
		'trailingAdvance': { enumerable: true, configurable: true, get: GlyphData.prototype.__get_trailingAdvance, set: GlyphData.prototype.__set_trailingAdvance}
	})
}


GlyphData.prototype.__get_num_glyphs = function() {
	var retVal = new FT_UInt("__skip_ptr__")
	retVal.__ptr = 	Polycode.GlyphData__get_num_glyphs(this.__ptr)
	return retVal
}

GlyphData.prototype.__set_num_glyphs = function(val) {
	Polycode.GlyphData__set_num_glyphs(this.__ptr, val.__ptr)
}

GlyphData.prototype.__get_trailingAdvance = function() {
	return Polycode.GlyphData__get_trailingAdvance(this.__ptr)
}

GlyphData.prototype.__set_trailingAdvance = function(val) {
	Polycode.GlyphData__set_trailingAdvance(this.__ptr, val)
}

Duktape.fin(GlyphData.prototype, function (x) {
	if (x === GlyphData.prototype) {
		return;
	}
	Polycode.GlyphData__delete(x.__ptr)
})

GlyphData.prototype.clearData = function() {
	Polycode.GlyphData_clearData(this.__ptr)
}
